packages/ai-core/src/browser/ai-configuration/template-settings-renderer.tsx

```diff
@@ -14,26 +14,103 @@
 // SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
 // *****************************************************************************
 import * as React from '@theia/core/shared/react';
-import { PromptCustomizationService } from '../../common/prompt-service';
-import { PromptTemplate } from '../../common';
+import { PromptCustomizationService, PromptService } from '../../common/prompt-service';
+import { AISettingsService, PromptTemplate } from '../../common';
 
-export interface TemplateSettingProps {
+const DEFAULT_VARIANT = 'default';
+
+export interface TemplateRendererProps {
     agentId: string;
     template: PromptTemplate;
     promptCustomizationService: PromptCustomizationService;
+    promptService: PromptService;
+    aiSettingsService: AISettingsService;
 }
 
-export const TemplateRenderer: React.FC<TemplateSettingProps> = ({ agentId, template, promptCustomizationService }) => {
+export const TemplateRenderer: React.FC<TemplateRendererProps> = ({
+    agentId,
+    template,
+    promptCustomizationService,
+    promptService,
+    aiSettingsService,
+}) => {
+    const [variantIds, setVariantIds] = React.useState<string[]>([]);
+    const [selectedVariant, setSelectedVariant] = React.useState<string>(DEFAULT_VARIANT);
+
+    React.useEffect(() => {
+        (async () => {
+            const variants = promptService.getVariantIds(template.id);
+            setVariantIds([DEFAULT_VARIANT, ...variants]);
+
+            const agentSettings = await aiSettingsService.getAgentSettings(agentId);
+            const currentVariant =
+                agentSettings?.selectedVariants?.[template.id] || DEFAULT_VARIANT;
+            setSelectedVariant(currentVariant);
+        })();
+    }, [template.id, promptService, aiSettingsService, agentId]);
+
+    const handleVariantChange = async (event: React.ChangeEvent<HTMLSelectElement>) => {
+        const newVariant = event.target.value;
+        setSelectedVariant(newVariant);
+
+        const agentSettings = await aiSettingsService.getAgentSettings(agentId);
+        const selectedVariants = agentSettings?.selectedVariants || {};
+
+        const updatedVariants = { ...selectedVariants };
+        if (newVariant === DEFAULT_VARIANT) {
```

Comment:
why do we need to handle that case special?

Reply:
I dont understand the question. If the default template is selected for a template ID, we remove the entry from the selected variants?

Reply:
yeah i see that, why not just store that the user selected the default variant?
I'm fine also with the current solution was just wondering, whether there is a special reason to do so

Reply:
Hmm, I find it cleaner to only store if variants are actually selected.